src/Runner.Worker/ActionManager.cs

```diff
@@ -100,43 +100,56 @@ public async Task<PrepareResult> PrepareActionsAsync(IExecutionContext execution
+                                                                        displayName: $"Pull {imageToPull.Key}",
+                                                                        data: new ContainerSetupInfo(imageToPull.Value, imageToPull.Key)));
+                        }
                     }
-                }
 
-                if (result.ImagesToBuild.Count > 0)
-                {
-                    foreach (var imageToBuild in result.ImagesToBuild)
+                    if (result.ImagesToBuild.Count > 0)
                     {
-                        var setupInfo = result.ImagesToBuildInfo[imageToBuild.Key];
-                        Trace.Info($"{imageToBuild.Value.Count} steps need to build image from '{setupInfo.Dockerfile}'");
-                        containerSetupSteps.Add(new JobExtensionRunner(runAsync: this.BuildActionContainerAsync,
-                                                                    condition: $"{PipelineTemplateConstants.Success}()",
-                                                                    displayName: $"Build {setupInfo.ActionRepository}",
-                                                                    data: new ContainerSetupInfo(imageToBuild.Value, setupInfo.Dockerfile, setupInfo.WorkingDirectory)));
+                        foreach (var imageToBuild in result.ImagesToBuild)
+                        {
+                            var setupInfo = result.ImagesToBuildInfo[imageToBuild.Key];
+                            Trace.Info($"{imageToBuild.Value.Count} steps need to build image from '{setupInfo.Dockerfile}'");
+                            containerSetupSteps.Add(new JobExtensionRunner(runAsync: this.BuildActionContainerAsync,
+                                                                        condition: $"{PipelineTemplateConstants.Success}()",
+                                                                        displayName: $"Build {setupInfo.ActionRepository}",
+                                                                        data: new ContainerSetupInfo(imageToBuild.Value, setupInfo.Dockerfile, setupInfo.WorkingDirectory)));
+                        }
                     }
-                }
 
 #if !OS_LINUX
-                if (containerSetupSteps.Count > 0)
-                {
-                    executionContext.Output("Container action is only supported on Linux, skip pull and build docker images.");
-                    containerSetupSteps.Clear();
-                }
+                    if (containerSetupSteps.Count > 0)
+                    {
+                        executionContext.Output("Container action is only supported on Linux, skip pull and build docker images.");
+                        containerSetupSteps.Clear();
+                    }
 #endif
+                }
+                return new PrepareResult(containerSetupSteps, result.PreStepTracker);
+
+            }
+            catch (Exception)
```

Comment:
Should we remove this try catch? It isn't doing anything besides catching and rethrowing right?

Reply:
+1

Reply:
I think we should use the `try...catch` patterns in scenarios where we need to do something with the exception.

Examples of that could be
- Logging
- Bubbling up this as an infra error (most relevant in this case)
- Checking if the exception is of a certain type, and if so ignoring it, otherwise re-throwing it

I'm not sure what value a catch block with `throw` as the only command buys it, I think

```
throw new Exception("test")
```

and 

``` 
try {
  throw new Exception("test")
} catch {
throw;
} 
```

functionally do the same exact thing, but one is less lines of code (and maybe more readable?)